stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/StreamFileUtil.kt

```diff
@@ -54,4 +59,127 @@ public object StreamFileUtil {
+     * Creates a Stream cache directory if one doesn't exist already.
+     *
+     * @param context The [Context] necessary to perform
+     * file operations.
+     * @param onError Called when an error occurs.
+     *
+     * @return The Stream cache directory if it exists or its creation
+     * was successful, false otherwise.
+     */
+    @Suppress("TooGenericExceptionCaught")
+    private fun getOrCreateStreamCacheDir(
+        context: Context,
+        onError: (exception: Exception) -> Unit = {},
+    ): File? {
+        return try {
+            val streamCacheDir = File(context.cacheDir, streamCacheDirName)
+            streamCacheDir.mkdir()
+
+            streamCacheDir
+        } catch (e: Exception) {
+            onError(e)
+            null
+        }
+    }
+
+    /**
+     * Deletes all the content contained within the
+     * Stream cache directory.
+     *
+     *  @param context The [Context] necessary to perform
+     *  file operations.
+     * @param onError Called when an error occurs.
+     *
+     * @return True if the cache was cleared, false otherwise.
+     */
+    @Suppress("TooGenericExceptionCaught")
+    public fun clearStreamCache(
+        context: Context,
+        onError: (exception: Exception) -> Unit = {},
+    ): Boolean {
+        return try {
+            val directory = File(context.cacheDir, streamCacheDirName)
+
+            directory.listFiles()?.forEach {
+                it.delete()
+            }
+
+            true
+        } catch (e: Exception) {
+            onError(e)
+            false
+        }
+    }
+
+    /**
+     * Hashes the links of given attachments and then tries to create a new file
+     * under that hash. If the file already exists checks that the full file
+     * has been written and shares it if it has, in other cases downloads the file
+     * and writes it.
+     *
+     * @param context The Android [Context] used for path resolving and [Uri] fetching.
+     * @param attachment the attachment to be downloaded.
+     * @param onError Called when an error occurs.
+     *
+     * @return The [Uri] that represents the path to the downloaded file.
+     */
+    public suspend fun writeFileToShareableFile(
+        context: Context,
+        attachment: Attachment,
+        onError: (exception: Exception) -> Unit = {},
+    ): Uri? {
+        val result = runCatching {
+            val streamCacheDir = getOrCreateStreamCacheDir(context, onError)
+            val attachmentName = (attachment.url ?: attachment.assetUrl)?.hashCode()
+            val fileName = attachmentName.toString() + attachment.name
+
+            val file = File(streamCacheDir, fileName)
+
+            // When File.createNewFile returns false it means that the file already exists.
+            // We then check the hash name equality to confirm it's the same file and check file size
+            // equality to make sure we've completed the download successfully.
+            if (!file.createNewFile() &&
```

Comment:
Unfortunately, the BE currently resizes images without updating the attachment file size, this means that this method is not reliable.

However, they have been made aware of this issue and will update the field in the future.